src/Algebra/Graph/Labelled/AdjacencyMap.hs

```diff
@@ -59,10 +59,8 @@ import qualified Data.Map.Strict as Map
 import qualified Data.Set        as Set
 
 -- | Edge-labelled graphs, where the type variable @e@ stands for edge labels.
--- For example, 'AdjacencyMap' @Bool@ @a@ is isomorphic to unlabelled graphs
--- defined in the top-level module "Algebra.Graph.AdjacencyMap", where @False@
--- and @True@ denote the lack of and the existence of an unlabelled edge,
--- respectively.
+-- For example, 'AdjacencyMap' @()@ @a@ is isomorphic to unlabelled graphs
```

Comment:
I don't like that this module becomes so different from `Algebra.Graph.Labelled` but not sure what to do about this. Something doesn't feel right, still.

Reply:
I agree.  Something still feels off.

I think the weirdness comes down to the fact that the edge type, really, _is_ a monoid, even if it's only declared as a semigroup.  This is because any semigroup can be trivially lifted to a monoid by wrapping it with `Maybe`, or, more generally, by defining a new, separate empty value and properly updating the binary operator to treat it appropriately.  In the case of these adjacency maps, we're doing exactly this: we're treating the _absence of an edge_ as `mempty`, thus forming an ad hoc monoid.

So, I think the reason the above text feels off is that, morally, we're constructing `AdjacencyMap (Maybe ()) a` as isomorphic to unlabelled graphs, but we're totally hiding the `Maybe` part as an implementation detail.  (To be clear, this "hiding Maybe as an implementation detail" is the whole reason I proposed this change in the first place.)  This should probably be called out in this comment.